src/app/user/register/post.controller.js

```diff
@@ -56,54 +57,62 @@ module.exports = (req, res) => {
   logger.info('Validating registration input');
   validator.validateChains([userChain, confirmuserChain, fnameChain, lnameChain])
     .then(() => {
-      logger.info('Starting whitelist check');
-      whitelist.isWhitelisted(usrname)
-        .then((result) => {
-          if (result) {
-            logger.info('Creating the user in the db');
-            userCreateApi.post(fname, lname, usrname, cookie.getInviteUserToken())
-              .then((dbUser) => {
-                if (Object.prototype.hasOwnProperty.call(JSON.parse(dbUser), 'message')) {
-                  logger.info('Failed to register user in db');
-                  logger.info(`${JSON.parse(dbUser).message}`);
-                  cookie.setUserEmail(null);
-                  return res.redirect('/user/regmsg');
-                }
-                const { userId } = JSON.parse(dbUser);
-                cookie.setUserDbId(userId);
-                logger.info('Calling gov notify service');
-                sendTokenService.send(fname, usrname, token)
-                  .then(() => {
-                    logger.info('Storing token in db');
-                    tokenApi.setToken(hashtoken, userId);
-                    res.redirect('/user/regmsg');
-                  })
-                  .catch((err) => {
-                    logger.error(`Failed to send notify email for ${usrname}`);
-                    logger.error(err);
-                    res.render('app/user/register/index', { cookie, errors: [regFailureError] });
-                  });
-              })
-              .catch((err) => {
-                logger.error(`Failed to create ${usrname} in DB`);
-                logger.error(err);
-                res.render('app/user/register/index', { cookie, errors: [regFailureError] });
-              });
-          } else {
-            // Not whitelisted
+      if (config.WHITELIST_REQUIRED) {
```

Comment:
You need to cast the env var from string to bool at some point. Otherwise you're checking if('true') and if('false') rather than if(true) and if(false)